test/framework/src/main/java/org/elasticsearch/test/rest/VersionBasedNodesFeatures.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.rest;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.core.Tuple;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// TODO[lor]: this needs to be replaced with (historical) feature checks once we have figured out a way of exposing them
+// to rest tests (e.g. through an internal API or similar)
+class VersionBasedNodesFeatures {
+    static class VersionBasedNodesFeaturesBuilder {
+        private final List<List<Boolean>> nodesFeaturesBitmap = new ArrayList<>();
```

Comment:
Why not use BitSet?

Reply:
Actually, I'm not sure using booleans gives us much here, there's only going to be O(tens) of features, so a simpler more direct representation is probably better here.